### ISSUE TITLE
get_last_insert_id fails when adding related item

### ISSUE THREAD

**Issue by sentinelweb** [sentinelweb/cuer#403]
- [x] Refactor to use GUID IDs in database, domain, net modules
- [x] update shared module to use GUIDs
- [x] update android app to use GUIDs
- [x] update ios app to use GUIDs
- [x] modify backup/restore to convert long ids to GUIDs 
- [x] Add playlist identifier to domain objects
- [x] modify dbinit JSON - maybe dont need it just make default and and a default yt playlist which can be imported
- [x] fix swipe add related item
- [x] check swipe add from search
- [x] bug: list is messed up after swipe move
- [x] bug: current playlist isn't restored on app restart
- [x] bug: custom actions for app playlists not working
- [x] bug: add playlist actionitem (on playlists) crashes
- [x] fix remaining tests
- [x] reenable backup check (maybe add timing blocker)
- [x] give list id descriptive string (dont need guid - just unique string)
- [x] add padding to bottom of setting screen (botom hidden under player)
- [x] fix flaky `QueueMediatorTest` (ignored!)
- [x] remove `TestCoroutineDispatcher` warnings for tests
- [x] crash when using prev button in queue (see below) - 

when moving item from the related playlist - media is saved but getLastInsetId fails so the playlist item creation fails.

Some efforts to debug/resolve are on branch : `feature/382-playlist-mvi-refactor-related-item-issue` - nothing worked though.

can just select the created media item but the (platform & platform_id) but it would be good to understand why `get_last_insert_id` isn't working as it should AFAIK. 

- Possibly another connection is created?

```
2023-01-11 22:00:17.752 7091-7091/uk.co.sentinelweb.cuer.app.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: uk.co.sentinelweb.cuer.app.debug, PID: 7091
    uk.co.sentinelweb.cuer.app.orchestrator.OrchestratorContract$DatabaseException: couldn't save playlist item
        at uk.co.sentinelweb.cuer.app.orchestrator.PlaylistItemOrchestrator.save(PlaylistItemOrchestrator.kt:145)
        at uk.co.sentinelweb.cuer.app.orchestrator.PlaylistItemOrchestrator$save$1.invokeSuspend(Unknown Source:15)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at android.os.Handler.handleCallback(Handler.java:942)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7898)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
    	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@f5a26c4, Dispatchers.Main]
    Caused by: java.lang.IllegalStateException: Save media failed vTyojqbW6lM
        at uk.co.sentinelweb.cuer.db.repository.SqldelightPlaylistItemDatabaseRepository$save$2.invokeSuspend(SqldelightPlaylistItemDatabaseRepository.kt:52)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
        at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
```

**Comment by sentinelweb** [comment]
﻿﻿2023-01-23 11:20:17.398 17793-17877/uk.co.sentinelweb.cuer.app.debug D/SqldelightMediaDatabaseRepository: save mediaEntity: 0 Great Minds - Introduction to the Problems and Scope of Philosophy
2023-01-23 11:20:17.404 17793-17877/uk.co.sentinelweb.cuer.app.debug D/SqldelightMediaDatabaseRepository: created mediaEntity: channelId: 123 Great Minds - Introduction to the Problems and Scope of Philosophy
2023-01-23 11:20:17.405 17793-17877/uk.co.sentinelweb.cuer.app.debug D/SqldelightMediaDatabaseRepository: created mediaEntity id: 9765
2023-01-23 11:20:17.405 17793-17877/uk.co.sentinelweb.cuer.app.debug D/SqldelightMediaDatabaseRepository: saved media:  9765 8ZoQ7wh9pSQ
2023-01-23 11:20:17.407 17793-17877/uk.co.sentinelweb.cuer.app.debug D/SqldelightMediaDatabaseRepository: loaded media:  9765 9765}
2023-01-23 11:20:17.474 17793-17793/uk.co.sentinelweb.cuer.app.debug W/View: requestLayout() improperly called by androidx.appcompat.view.menu.ActionMenuItemView{4447408 VFED..C.. ......ID 0,14-168,182 #7f0a02b0 app:id/playlist_help} during layout: running second layout pass
2023-01-23 11:20:17.474 17793-17793/uk.co.sentinelweb.cuer.app.debug W/View: requestLayout() improperly called by androidx.appcompat.view.menu.ActionMenuItemView{2b007a1 VFED..C.. ......ID 168,14-336,182 #7f0a02ba app:id/playlist_view_cards} during layout: running second layout pass
2023-01-23 11:20:17.474 17793-17793/uk.co.sentinelweb.cuer.app.debug W/View: requestLayout() improperly called by androidx.appcompat.view.menu.ActionMenuItemView{ecc7dc6 VFED..C.. ......ID 336,14-504,182 #7f0a02b4 app:id/playlist_search} during layout: running second layout pass
2023-01-23 11:20:17.475 17793-17793/uk.co.sentinelweb.cuer.app.debug W/View: requestLayout() improperly called by androidx.appcompat.view.menu.ActionMenuItemView{8881187 VFED..C.. ......ID 504,14-672,182 #7f0a01d6 app:id/menu_paste_add} during layout: running second layout pass
2023-01-23 11:20:17.475 17793-17793/uk.co.sentinelweb.cuer.app.debug W/View: requestLayout() improperly called by androidx.appcompat.view.menu.ActionMenuItemView{cb0c8b4 VFED..C.. ......ID 672,14-840,182 #7f0a01d8 app:id/menu_settings} during layout: running second layout pass
2023-01-23 11:20:17.476 17793-17793/uk.co.sentinelweb.cuer.app.debug W/View: requestLayout() improperly called by androidx.appcompat.widget.ActionMenuView{3a47cdd V.E...... ......ID 600,0-1440,196} during layout: running second layout pass
2023-01-23 11:20:17.486 17793-17793/uk.co.sentinelweb.cuer.app.debug D/PlaylistMviModelMapper: map
2023-01-23 11:20:17.500 17793-17793/uk.co.sentinelweb.cuer.app.debug D/PlaylistMviStoreFactory: media changed: FULL, LOCAL, id=9765 title=I Read 50 Philosophy Books: Here's What I Learned
2023-01-23 11:20:17.500 17793-17877/uk.co.sentinelweb.cuer.app.debug E/SqldelightPlaylistItemDatabaseRepository: couldn't save playlist item
    android.database.sqlite.SQLiteConstraintException: UNIQUE constraint failed: playlist_item.media_id, playlist_item.playlist_id (code 2067 SQLITE_CONSTRAINT_UNIQUE)
        at android.database.sqlite.SQLiteConnection.nativeExecute(Native Method)
        at android.database.sqlite.SQLiteConnection.execute(SQLiteConnection.java:730)
        at android.database.sqlite.SQLiteSession.execute(SQLiteSession.java:621)
        at android.database.sqlite.SQLiteStatement.execute(SQLiteStatement.java:47)
        at androidx.sqlite.db.framework.FrameworkSQLiteStatement.execute(FrameworkSQLiteStatement.java:41)
        at com.squareup.sqldelight.android.AndroidPreparedStatement.execute(AndroidSqliteDriver.kt:220)
        at com.squareup.sqldelight.android.AndroidSqliteDriver$execute$2.invoke(AndroidSqliteDriver.kt:142)
        at com.squareup.sqldelight.android.AndroidSqliteDriver$execute$2.invoke(AndroidSqliteDriver.kt:142)
        at com.squareup.sqldelight.android.AndroidSqliteDriver.execute(AndroidSqliteDriver.kt:127)
        at com.squareup.sqldelight.android.AndroidSqliteDriver.execute(AndroidSqliteDriver.kt:142)
        at uk.co.sentinelweb.cuer.app.db.database.PlaylistItemEntityQueriesImpl.create(DatabaseImpl.kt:1847)
        at uk.co.sentinelweb.cuer.db.repository.SqldelightPlaylistItemDatabaseRepository$save$2.invokeSuspend(SqldelightPlaylistItemDatabaseRepository.kt:65)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.UndispatchedCoroutine.afterResume(CoroutineContext.kt:233)
        at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
        at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
2023-01-23 11:20:17.513 17793-17793/uk.co.sentinelweb.cuer.app.debug D/PlaylistMviModelMapper: map
2023-01-23 11:20:17.515 17793-17793/uk.co.sentinelweb.cuer.app.debug D/PlaylistMviModelMapper: map
2023-01-23 11:20:17.518 17793-17793/uk.co.sentinelweb.cuer.app.debug D/PlaylistMviModelMapper: map
2023-01-23 11:20:17.524 17793-17793/uk.co.sentinelweb.cuer.app.debug E/Cuer: FATAL Global Exception
    uk.co.sentinelweb.cuer.app.orchestrator.OrchestratorContract$DatabaseException: couldn't save playlist item
        at uk.co.sentinelweb.cuer.app.orchestrator.PlaylistItemOrchestrator.save(PlaylistItemOrchestrator.kt:145)
        at uk.co.sentinelweb.cuer.app.orchestrator.PlaylistItemOrchestrator$save$1.invokeSuspend(Unknown Source:15)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at android.os.Handler.handleCallback(Handler.java:942)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7898)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
    	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@8ce48aa, Dispatchers.Main]
    Caused by: android.database.sqlite.SQLiteConstraintException: UNIQUE constraint failed: playlist_item.media_id, playlist_item.playlist_id (code 2067 SQLITE_CONSTRAINT_UNIQUE)
        at android.database.sqlite.SQLiteConnection.nativeExecute(Native Method)
        at android.database.sqlite.SQLiteConnection.execute(SQLiteConnection.java:730)
        at android.database.sqlite.SQLiteSession.execute(SQLiteSession.java:621)
        at android.database.sqlite.SQLiteStatement.execute(SQLiteStatement.java:47)
        at androidx.sqlite.db.framework.FrameworkSQLiteStatement.execute(FrameworkSQLiteStatement.java:41)
        at com.squareup.sqldelight.android.AndroidPreparedStatement.execute(AndroidSqliteDriver.kt:220)
        at com.squareup.sqldelight.android.AndroidSqliteDriver$execute$2.invoke(AndroidSqliteDriver.kt:142)
        at com.squareup.sqldelight.android.AndroidSqliteDriver$execute$2.invoke(AndroidSqliteDriver.kt:142)
        at com.squareup.sqldelight.android.AndroidSqliteDriver.execute(AndroidSqliteDriver.kt:127)
        at com.squareup.sqldelight.android.AndroidSqliteDriver.execute(AndroidSqliteDriver.kt:142)
        at uk.co.sentinelweb.cuer.app.db.database.PlaylistItemEntityQueriesImpl.create(DatabaseImpl.kt:1847)
        at uk.co.sentinelweb.cuer.db.repository.SqldelightPlaylistItemDatabaseRepository$save$2.invokeSuspend(SqldelightPlaylistItemDatabaseRepository.kt:65)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.UndispatchedCoroutine.afterResume(CoroutineContext.kt:233)
        at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
        at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)

**Comment by sentinelweb** [comment]
fixed on #414
